test/integration/node-specific/ipv6.test.ts

```diff
@@ -0,0 +1,94 @@
+import { expect } from 'chai';
+import * as net from 'net';
+import * as process from 'process';
+import * as sinon from 'sinon';
+
+import { ConnectionCreatedEvent, MongoClient, ReadPreference, TopologyType } from '../../../src';
+import { byStrings, sorted } from '../../tools/utils';
+
+describe('IPv6 Addresses', () => {
+  let client: MongoClient;
+  let ipv6Hosts: string[];
+
+  beforeEach(async function () {
+    if (
+      process.platform !== 'win32' ||
+      this.configuration.topologyType !== TopologyType.ReplicaSetWithPrimary
+    ) {
+      if (this.currentTest) {
+        // Ubuntu 18 does not support localhost AAAA lookups (IPv6)
+        // Windows (VS2019) has the AAAA lookup
+        // We do not run a replica set on macos
```

Comment:
if we don't run replica sets on macos, then this would already be skipped on macos by the topologyType check, no? I think it might make more sense to exclude the platform that does not support macos AAAA lookups than to selectively only include windows because that's the one that has the replica set and the support. Or am I missing something?

Reply:
Nope that's correct, this would work on MacOS the following tests just depend on being in a replica set. Changed!